src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -159,6 +159,7 @@ protected override Size MeasureOverride(Size availableSize)
 
             try
             {
+                _realizedElements?.ValidateStartU(Orientation);
```

Comment:
This relies on the containers being re-measured before the `VirtualizingStackPanel` is remeasured. This works in the general case, but not in all cases.

Reply:
This should have been addressed with the latest changes.